lib/ebae/accounts/credential.ex

```diff
@@ -0,0 +1,30 @@
+defmodule Ebae.Accounts.Credential do
+  use Ecto.Schema
+  import Ecto.Changeset
+  alias Ebae.Accounts.User
+
+  schema "credentials" do
+    field :email, :string
+    field :password, :string
+    belongs_to :user, User
+
+    timestamps()
+  end
+
+  @doc false
+  def changeset(credential, attrs) do
+    credential
+    |> cast(attrs, [:email, :password])
+    |> validate_required([:email, :password])
+    |> unique_constraint(:email)
+    |> put_password_hash()
+  end
+
+  defp put_password_hash(
+         %Ecto.Changeset{valid?: true, changes: %{password: password}} = changeset
+       ) do
+    change(changeset, password: Pbkdf2.hash_pwd_salt(password))
```

Comment:
Why `Pbkdf2` and not bcrypt?

Reply:
Switched over to Argon2 for improved security.